src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/storage.go

```diff
@@ -127,6 +128,12 @@ func (d *Database) CreateAccount(
 		return nil, err
 	}
 	if err := d.profiles.insertProfile(ctx, localpart); err != nil {
+		if err, ok := err.(*pq.Error); ok {
+			if err.Code.Class() == "23" {
+				// 23 => unique_violation => Account already exists
+				return nil, nil
+			}
+		}
```

Comment:
I would be tempted to pull this out into a function in `common/sql.go`, something like `IsConstraintViolationErr(err error)`

Reply:
I just realized `err.Code.Class() == 23` represents any constraint violation.

In this case, I think we'd rather have a `IsUniqueConstraintViolationErr(err error)` that checks if the error code is `23505`, since we want to return `nil,nil` only if the account already exists